academy/ibc-dev/ibcapp_checkers.md

```diff
@@ -0,0 +1,525 @@
+        }
+        k.MustPayWinnings(ctx, &storedGame)
+
+        // Here you can register a draw
+        k.MustRegisterPlayerForfeit(ctx, &storedGame)
+
+        storedGame.Board = ""
+        k.SetStoredGame(ctx, storedGame)
+    }
+    
+    ...
+```
+
+That will get the job done and add the player win, loss, or forfeit counts to the store.
+
+It is time to sort the players and clip the leaderboard to the best 100(`LeaderboardWinnerLength`) players. Thus, scaffold a new transaction:
+
+```bash
+$ ignite scaffold message updateBoard --module leaderboard
+```
+
+Again, you can first create some helper functions in `x/leaderboard/keeper/board.go`:
+
+
+```golang
+...
+
+    func ParseDateAddedAsTime(dateAdded string) (dateAddedParsed time.Time, err error) {
+        dateAddedParsed, errDateAdded := time.Parse(types.TimeLayout, dateAdded)
+        return dateAddedParsed, sdkerrors.Wrapf(errDateAdded, types.ErrInvalidDateAdded.Error(), dateAdded)
+    }
+
+    func SortPlayerInfo(playerInfoList []types.PlayerInfo) {
+        sort.SliceStable(playerInfoList[:], func(i, j int) bool {
+            if playerInfoList[i].WonCount > playerInfoList[j].WonCount {
+                return true
+            }
+            if playerInfoList[i].WonCount < playerInfoList[j].WonCount {
+                return false
+            }
+            firstPlayerTime, _ := ParseDateAddedAsTime(playerInfoList[i].DateUpdated)
+            secondPlayerTime,_ := ParseDateAddedAsTime(playerInfoList[j].DateUpdated)
+
+            return firstPlayerTime.After(secondPlayerTime)
+        })
+    }
+
+    func (k Keeper) updateBoard(ctx sdk.Context, playerInfoList []types.PlayerInfo) {
+        SortPlayerInfo(playerInfoList)
+
+        if types.LeaderboardWinnerLength < uint64(len(playerInfoList)) {
+            playerInfoList = playerInfoList[:types.LeaderboardWinnerLength]
+        }
+
+        k.SetBoard(ctx, types.Board {
+            PlayerInfo: playerInfoList,
+        })
+    }
+```
+
+If it cannot parse the date information, it will throw an error that you need to include in `x/leaderboard/types/errors.go`:
+
+```golang
+    ErrInvalidDateAdded     = sdkerrors.Register(ModuleName, 1120, "dateAdded cannot be parsed: %s")
+```
+
+And you need to call `updateBoard` in `x/leaderboard/keeper/msg_server_update_board.go`:
+
+```golang
+package keeper
+
+import (
+    "context"
+
+    "github.com/b9lab/checkers/x/leaderboard/types"
+    sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k msgServer) UpdateBoard(goCtx context.Context, msg *types.MsgUpdateBoard) (*types.MsgUpdateBoardResponse, error) {
+    ctx := sdk.UnwrapSDKContext(goCtx)
+
+    playerInfoList := k.GetAllPlayerInfo(ctx)
+    k.updateBoard(ctx, playerInfoList)
+
+    return &types.MsgUpdateBoardResponse{}, nil
+}
+```
+
+That is it! Now the checkers blockchain can keep track of the player information, and create or update the leaderboard based on the player information if requested via the CLI.
```

Comment:
I'd make a split here and start a new section/file to discuss IBC specific parts vs the prep that is more checkers/SDK focused